### ISSUE TITLE
[a11y]: Contained List [all variants] - list title which should be marked as heading

### ISSUE THREAD

**Issue by andreancardona** [carbon-design-system/carbon#13940]
### Package

@carbon/react

### Browser
FF

### Package version

carbon/react@1.30.0

### Automated testing tool and ruleset
Manual QA

### Assistive technology

JAWS & VO

### Description

The list title is not marked up semantically as heading

When the semantic structure of the element is not conveyed programmatically, Blind screen reader user would not be able to perceive it

### WCAG 2.1 Violation

1.3.1 Info and Relationships 

### Reproduction/example

Record `Number 28` in the below table:
https://airtable.com/shrFsaeHhImOkuVNR

### Steps to reproduce
The following text visually looks prominent like heading. However SR does not announce this as heading. Only announced as text. "List title". 

Note: This issue appears in all the instances where the title text is present.

### Recommended fix
Assign appropriate heading level 

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by andreancardona** [comment]
Confirmed this doesn't need to be a `heading` currently we have `aria-labelledby` to proved the `title` it's needed context to the list item elements 

Reviewed with @mbgower 🎉
